packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -86,6 +86,53 @@ All email subjects, bodies and SMS messages for both invitation and verification
 Learn more about [message templates
 here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-message-templates.html).
 
+### Sign In
+
+Users registering or signing in into your application can do so with multiple identifiers. There are 4 options
+available:
+
+* `USERNAME`: Allow signing in using the one time immutable user name that the user chose at the time of sign up.
+* `EMAIL`: Allow signing in using the email address that is associated with the account.
+* `PHONE_NUMBER`: Allow signing in using the phone number that is associated with the account.
+* `PREFERRED_USERNAME`: Allow signing in with an alternate user name that the user can change at any time. However, this
+  is not available if the USERNAME option is not chosen.
+
+The following code sets up a user pool so that the user can sign in with either their username or their email address -
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  signInAliases: [ SignInAlias.USERNAME, SignInAlias.EMAIL ],
+});
+```
+
+User pools can either be configured so `USERNAME` is primary sign in form, but also allows for the other three to be
+used additionally; or it can be configured so that email and/or phone numbers are the only ways a user can register and
+sign in. Read more about this
+[here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-aliases-settings).
+
+To match with 'Option 1' in the above link, with a verified email, this property should be set to
+`[ UsernameAlias.USERNAME, UsernameAlias.EMAIL ]`. To match with 'Option 2' in the above link with both a verified
+email and phone number, this property should be set to `[ UsernameAlias.EMAIL, UsernameAlias.PHONE ]`.
+
+Cognito recommends that email and phone number be automatically verified, if they are one of the sign in methods for
+the user pool. Read more about that
+[here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-aliases).
+The CDK does this by default, when email and/or phone number are specified as part of `signInAliases`. This can be
+overridden by specifying the `autoVerifiedAttributes` property.
+
+The following code snippet sets up only email as a sign in alias, but both email and phone number to be auto-verified.
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  signInAliases: [ SignInAlias.USERNAME, SignInAlias.EMAIL ],
+  autoVerifiedAttributes: [ AutoVerifiedAttrs.EMAIL, AutoVerifiedAttrs.PHONE ]
```

Comment:
Can also be nice to use a set of boolean flags for this one.

Maybe rename `autoVerifiedAttributes` to `autoVerify` (conveys the same meaning).